runtime/src/main/java/org/corfudb/runtime/view/EpochedClient.java

```diff
@@ -0,0 +1,155 @@
+package org.corfudb.runtime.view;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nonnull;
+
+import lombok.Data;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.BaseClient;
+import org.corfudb.runtime.clients.IClient;
+import org.corfudb.runtime.clients.IClientRouter;
+import org.corfudb.runtime.clients.LayoutClient;
+import org.corfudb.runtime.clients.LogUnitClient;
+import org.corfudb.runtime.clients.ManagementClient;
+import org.corfudb.runtime.clients.SequencerClient;
+import org.corfudb.runtime.exceptions.QuorumUnreachableException;
+import org.corfudb.runtime.exceptions.WrongEpochException;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.corfudb.runtime.view.Layout.LayoutSegment;
+
+/**
+ * This is a wrapper over the layout to provide the clients required to communicate with the nodes.
+ * <p>Created by zlokhandwala on 3/8/18.
+ */
+@Slf4j
+@Data
+public class EpochedClient {
+
+    @Getter
+    private final Layout layout;
+
+    /**
+     * The org.corfudb.runtime this layout is associated with.
+     */
+    @Getter
+    private final CorfuRuntime runtime;
+
+    /**
+     * Defensive constructor since we can create a Layout from a JSON file.
+     * JSON deserialize is forced through this constructor.
+     */
+    public EpochedClient(@Nonnull Layout layout, @Nonnull CorfuRuntime corfuRuntime) {
+
+        this.layout = new Layout(layout);
+        this.runtime = corfuRuntime;
+    }
+
+    /**
+     * Attempts to move all servers in the system to the epoch of this layout.
+     * The seal however waits only for a response from a quorum of layout servers (n/2 + 1),
+     * a quorum of log unit servers in every stripe in case of QUORUM_REPLICATION or
+     * at least one log unit server in every stripe in case of CHAIN_REPLICATION.
+     *
+     * @throws WrongEpochException        If any server is in a higher epoch.
+     * @throws QuorumUnreachableException If enough number of servers cannot be sealed.
+     */
+    public void moveServersToEpoch()
+            throws WrongEpochException, QuorumUnreachableException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.runtime.exceptions.WrongEpochException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.runtime.exceptions.QuorumUnreachableException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)